cadence/contracts/GametvNFT.cdc

```diff
@@ -226,7 +231,14 @@ pub contract GametvNFT: NonFungibleToken {
 
             // Increment count for NFT of particular type
             nftType.updateCount(count: updateCount)
-            GametvNFT.nftTypes[typeId] = nftType
+
+            if (updateCount < nftType.maxCount) {
+                GametvNFT.activeNftTypes[typeId] = nftType
+            } else {
+                GametvNFT.historicNftTypes[typeId] = nftType
+                GametvNFT.activeNftTypes.remove(key: typeId)
+                emit TypeMoved(typeId: typeId)
```

Comment:
Can make the event name more descriptive: instead of TypeMoved, it can be TypeSoldOut perhaps.